extensions/zacoons/webgpu.js

```diff
@@ -0,0 +1,638 @@
+                defaultValue: "",
+              },
+              DST: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: "",
+              },
+            },
+          },
+          {
+            opcode: "createCommandBuffer",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "🗍 create command buffer",
+            disableMonitor: true,
+          },
+
+          label("Device Queues"),
+          {
+            opcode: "newQueue",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "≫ new queue",
+          },
+          {
+            opcode: "writeListToBuffer",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "≫ write list [LIST_ID] with type [TYPE] to buffer [BUFFER]",
+            arguments: {
+              LIST_ID: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "lists",
+              },
+              TYPE: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "types",
+              },
+              BUFFER: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: "",
+              },
+            },
+          },
+          {
+            opcode: "addCommandBufferToQueue",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "≫ add command buffer [COMMAND_BUFFER] to queue",
+            arguments: {
+              COMMAND_BUFFER: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: "",
+              },
+            },
+          },
+          {
+            opcode: "submitQueue",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "≫ submit queue",
+          },
+        ],
+        menus: {
+          bindGroupLayoutEntryTypes: {
+            acceptReporters: true,
+            items: ["uniform", "read-only-storage", "storage"],
+          },
+          bufferUsages: {
+            acceptReporters: true,
+            items: [
+              "COPY_SRC",
+              "COPY_DST",
+              "INDEX",
+              "INDIRECT",
+              "MAP_READ",
+              "MAP_WRITE",
+              "QUERY_RESOLVE",
+              "STORAGE",
+              "UNIFORM",
+              "VERTEX",
+            ],
+          },
+          modules: {
+            acceptReporters: true,
+            items: "_getModuleNames",
+          },
+          lists: {
+            acceptReporters: true,
+            items: "_getLists",
+          },
+          types: {
+            acceptReporters: true,
+            items: Object.keys(this.types),
+          },
+        },
+      };
+    }
+
+    init() {
+      this._resetAttrs();
+
+      if (!navigator.gpu) return false;
+
+      return navigator.gpu.requestAdapter().then(async (adapter) => {
+        this.dev = await adapter.requestDevice();
+        this._createModules();
+        return true;
+      });

```

Comment:
Promise getting rejected does not throw an exception here, so wrapping it in `try {} catch {}` didn't fix it. The old code was fine, it just needed an extra `.catch()`.
```suggestion
      }).catch(() => false);
```

Reply:
oh yeah, beginner mistake, thanks